client/grpc-web-client/src/client.ts

```diff
@@ -264,7 +298,8 @@ class GrpcClient<TRequest extends ProtobufMessage, TResponse extends ProtobufMes
     requestHeaders.set("content-type", "application/grpc-web+proto");
     requestHeaders.set("x-grpc-web", "1"); // Required for CORS handling
 
-    this.transport.start(requestHeaders);
+    const middlewareHeaders = this.middleware.onStart(requestHeaders) || requestHeaders;
```

Comment:
IMHO it would be cleaner/easier to extend for the middleware.onStart interceptor to mutate the `Metadata` object instead of returning it; the `Metadata` object is, after-all fully mutable.